module/VuFind/src/VuFind/RecordDriver/SolrDefault.php

```diff
@@ -298,4 +301,74 @@ public function getContainerRecordID()
             && !empty($this->fields['hierarchy_parent_id'])
             ? $this->fields['hierarchy_parent_id'][0] : '';
     }
+
+    /**
+     * Get work identification keys
+     *
+     * @return array
+     */
+    public function getWorkKeys()
+    {
+        return $this->fields['work_keys_str_mv'] ?? [];
+    }
+
+    /**
+     * Return count of other versions available
+     *
+     * @return int
+     */
+    public function getOtherVersionCount()
+    {
+        if (null === $this->searchService) {
+            return false;
+        }
+
+        if (!($workKeys = $this->getWorkKeys())) {
+            return false;
+        }
+
+        if (!isset($this->otherVersionsCount)) {
+            $params = new \VuFindSearch\ParamBag();
+            $params->add('rows', 0);
+            $results = $this->searchService->workExpressions(
+                $this->getSourceIdentifier(),
+                $this->getUniqueID(),
+                $workKeys,
+                $params
+            );
+            $this->otherVersionsCount = $results->getTotal();
+        }
+        return $this->otherVersionsCount;
+    }
+
+    /**
+     * Retrieve versions as a search result
+     *
+     * @param bool $includeSelf Whether to include this record
+     * @param int  $count       Maximum number of records to display
+     *
+     * @return \VuFindSearch\Response\RecordCollectionInterface
+     */
+    public function getVersions($includeSelf = false, $count = 20)
+    {
+        if (null === $this->searchService) {
+            return false;
+        }
+
+        if (!($workKeys = $this->getWorkKeys())) {
+            return false;
+        }
+
+        if (!isset($this->otherVersions)) {
+            $params = new \VuFindSearch\ParamBag();
+            $params->add('rows', min($count, 100));
```

Comment:
Should this hard-coded limit be a property or constant?

Reply:
Made it a configurable property. No default in searches.ini since I don't anticipate great need to change it, but it's there just in case.

Reply:
Thanks!